java/src/main/java/com/google/appengine/tools/pipeline/impl/util/DIUtil.java

```diff
@@ -1,41 +1,52 @@
 package com.google.appengine.tools.pipeline.impl.util;
 
+/**
+ * Copyright Worklytics, Co. 2024.
+ */
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
-import dagger.ObjectGraph;
+
 import lombok.NonNull;
 import lombok.extern.java.Log;
 
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
 /**
- * Helper to get an ObjectGraph out of a Dagger Module.
+ * Helper to get a component instance out of Dagger, support runtime injection of classes.
  * <p>
- * This is mostly intended to be used for jobs, thus Default Module is linked somehow to AsyncModule.
  *
  */
-
 @Log
 public class DIUtil {
 
 
 	private static final Object lock = new Object();
 
-	private static final Map<String, ObjectGraph> objectGraphCache = new ConcurrentHashMap<>(1);
+	private static final Map<Class<?>, Object> componentCache = new ConcurrentHashMap<>(1);
 
-	private static final Map<String, ObjectGraph> overriddenObjectGraphCache = new ConcurrentHashMap<>(1);
+	private static final Map<Class<?>, Object> overriddenComponentCache = new ConcurrentHashMap<>(1);
 
 	// just for logging purposes
 	private static boolean overridden = false;
 
 	/**
 	 * Injects and Injectable instance through Reflection
+   * @param componentClass component class (Dagger-generated)
 	 * @param injectable class to inject
 	 */
-	public static void inject(String moduleClassFQN, Object injectable) {
-		ObjectGraph objectGraph = getFromModuleClass(moduleClassFQN);
-    objectGraph.inject(injectable);
+	public static void inject(Class<?> componentClass, Object injectable) {
```

Comment:
Why not 

```suggestion
	public static void inject(Class<?> componentClass, Injectable injectable) {
```
?

So it can fails on compile time ?

Reply:
`Injectable` is an annotation .. but yeah, will refactor to make things cleaer